cppsolutions/FindMaximumNumber.cpp

```diff
@@ -0,0 +1,121 @@
+#include <vector>
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+class FindMaximumNumber
+{
+private:
+
+	vector<int> testCases{
+		5555,
+		-5555,
+		1234,
+		-1234,
+		0,
+		556,
+		-556
+	};
+
+	int getTestCase(int idx)
+	{
+		return testCases[idx];
+	}
+
+	int findMaximumNumber(int number, int digit)
+	{
+		int numberOfDigits = number == 0 ? 0 : floor(log10(abs(number))) + 1;
+		string numberAsString = to_string(number);
+		bool negative = number < 0 ? true : false;
+		int outputNumber = 0;
+
+		if (negative)
+		{
+			int i = 1;
+			for (; i <= numberOfDigits; i++)
```

Comment:
Nitpick... Why can't we combine the initialization on same line 36

Reply:
Simply because I want i to be checked even after when the loop is finished. If i is initialized inside the loop, I need another variable to carry the final value of i that I wanted to avoid.